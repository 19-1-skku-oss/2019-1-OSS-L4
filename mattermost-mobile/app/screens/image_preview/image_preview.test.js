// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';
import {
    TouchableOpacity,
} from 'react-native';

import Preferences from 'mattermost-redux/constants/preferences';

import ImagePreview from './image_preview';

jest.useFakeTimers();
jest.mock('react-intl');
jest.mock('react-native-doc-viewer', () => {
    return {
        OpenFile: jest.fn(),
    };
});

describe('ImagePreview', () => {
    const baseProps = {
        canDownloadFiles: true,
        deviceHeight: 400,
        deviceWidth: 300,
        files: [
            {caption: 'Caption 1', source: 'source', data: 'data'},
            {caption: 'Caption 2', source: 'source', data: 'data'},
        ],
        getItemMeasures: jest.fn(),
        index: 0,
        navigator: {setStyle: jest.fn()},
        origin: {},
        target: {},
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ImagePreview {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(TouchableOpacity).first().exists()).toEqual(true);
    });

    test('should match snapshot, renderDownloadButton', () => {
        const wrapper = shallow(
            <ImagePreview {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.instance().renderDownloadButton()).toMatchSnapshot();
    });

    test('should match state on handleChangeImage', () => {
        const wrapper = shallow(
            <ImagePreview {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.setState({index: 0});
        wrapper.instance().handleChangeImage(1);

        expect(wrapper.state('index')).toEqual(1);
    });

    test('should match call getItemMeasures & navigator.setStyle on close', () => {
        const getItemMeasures = jest.fn();
        const navigator = {setStyle: jest.fn()};
        const wrapper = shallow(
            <ImagePreview
                {...baseProps}
                getItemMeasures={getItemMeasures}
                navigator={navigator}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.instance().close();

        expect(navigator.setStyle).toHaveBeenCalledTimes(2);
        expect(navigator.setStyle).toBeCalledWith({screenBackgroundColor: 'transparent'});
    });
});
