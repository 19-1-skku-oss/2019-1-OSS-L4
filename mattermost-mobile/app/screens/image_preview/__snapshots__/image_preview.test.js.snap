// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview should match snapshot 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "flex": 1,
      },
      Object {
        "opacity": NaN,
      },
    ]
  }
>
  <AnimatedComponent
    style={
      Object {
        "backgroundColor": "#000",
        "flex": 1,
      }
    }
  >
    <Gallery
      errorComponent={[Function]}
      flatListProps={
        Object {
          "windowSize": 3,
        }
      }
      imageComponent={[Function]}
      images={
        Array [
          Object {
            "caption": "Caption 1",
            "data": "data",
            "source": "source",
          },
          Object {
            "caption": "Caption 2",
            "data": "data",
            "source": "source",
          },
        ]
      }
      initialPage={0}
      onPageSelected={[Function]}
      onSingleTapConfirmed={[Function]}
      onSwipedVertical={[Function]}
      pageMargin={2}
      removeClippedSubviews={true}
      scrollViewStyle={Object {}}
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <AnimatedComponent
      style={
        Array [
          Object {
            "height": 48,
            "overflow": "hidden",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "opacity": 1,
            "top": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.8)",
            "height": 48,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 44,
                "justifyContent": "center",
                "width": 48,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#fff"
              name="md-close"
              size={26}
            />
          </TouchableOpacity>
          <Text
            style={
              Object {
                "color": "white",
                "flex": 1,
                "fontSize": 15,
                "marginHorizontal": 10,
                "textAlign": "center",
              }
            }
          >
            1/2
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 44,
                "justifyContent": "center",
                "width": 48,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#fff"
              name="ios-download"
              size={26}
            />
          </TouchableOpacity>
        </View>
      </View>
    </AnimatedComponent>
    <AnimatedComponent
      style={
        Array [
          Object {
            "bottom": 0,
            "opacity": 1,
          },
          Object {
            "height": 64,
            "justifyContent": "center",
            "overflow": "hidden",
            "position": "absolute",
            "width": "100%",
          },
        ]
      }
    >
      <LinearGradient
        colors={
          Array [
            "transparent",
            "#000000",
          ]
        }
        end={
          Object {
            "x": 0,
            "y": 0.9,
          }
        }
        pointerEvents="none"
        start={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 64,
            "justifyContent": "flex-end",
            "left": 0,
            "paddingBottom": 5,
            "paddingHorizontal": 24,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "marginBottom": 10,
            }
          }
        >
          Caption 1
        </Text>
      </LinearGradient>
    </AnimatedComponent>
  </AnimatedComponent>
  <Downloader
    deviceHeight={400}
    deviceWidth={300}
    file={
      Object {
        "caption": "Caption 1",
        "data": "data",
        "source": "source",
      }
    }
    force={false}
    onCancelPress={[Function]}
    onDownloadCancel={[Function]}
    onDownloadStart={[Function]}
    onDownloadSuccess={[Function]}
    prompt={false}
    saveToCameraRoll={true}
    show={false}
  />
</AnimatedComponent>
`;

exports[`ImagePreview should match snapshot, renderDownloadButton 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 44,
      "justifyContent": "center",
      "width": 48,
    }
  }
>
  <Icon
    allowFontScaling={false}
    color="#fff"
    name="ios-download"
    size={26}
  />
</TouchableOpacity>
`;
